@Test public void testFalse() throws Exception {
  assertTrue(!(new UnaryOr<Object>()).test("xyzzy"));
  assertTrue(!(new UnaryOr<Object>(Constant.FALSE)).test("xyzzy"));
  assertTrue(!(new UnaryOr<Object>(Constant.FALSE,Constant.FALSE)).test("xyzzy"));
  assertTrue(!(new UnaryOr<Object>(Constant.FALSE,Constant.FALSE,Constant.FALSE)).test("xyzzy"));
  UnaryOr<Object> p=new UnaryOr<Object>(Constant.FALSE);
  assertTrue(!p.test("xyzzy"));
  for (int i=0; i < 10; i++) {
    p.or(Constant.FALSE);
    assertTrue(!p.test("xyzzy"));
  }
  UnaryOr<Object> q=new UnaryOr<Object>(Constant.FALSE);
  assertTrue(!q.test("xyzzy"));
  for (int i=0; i < 10; i++) {
    q.or(Constant.FALSE);
    assertTrue(!q.test("xyzzy"));
  }
  UnaryOr<Object> r=new UnaryOr<Object>(p,q);
  assertTrue(!r.test("xyzzy"));
}
